web_server/src/routes/appointment.rs

```diff
@@ -0,0 +1,277 @@
+use super::*;
+use shared::domain::{Appointment, DBAppointment, NewAppointment, NewInvitation, InvitationParams, SendAppointmentEmails};
+use shared::qr_client::QRClient;
+use futures::future::join_all;
+use url::Url;
+use shared::email_client::EmailClient;
+
+pub fn appointment_routes(cfg: &mut web::ServiceConfig) {
+    cfg.service(
+        web::resource("/appointment")
+            .route(web::post().to(add_appointment))
+    )
+        .service(
+            web::resource("/appointment/{id}")
+                .route(web::get().to(get_appointment_by_id))
+                .route(web::delete().to(delete_appointment))
+        )
+        .service(
+            web::resource("/appointment/{id}/invitation")
+                .route(web::post().to(add_invitation))
+        );
+}
+
+
+#[tracing::instrument(
+    name = "Add invitation service handler",
+    skip(pool, qr_client, email_client),
+)]
+pub async fn add_invitation(
+    appointment_id: web::Path<Uuid>,
+    query_params: web::Query<InvitationParams>,
+    send_appointment_emails: web::Json<SendAppointmentEmails>,
+    pool: Data<PgPool>,
+    qr_client: Data<QRClient>,
+    email_client: Data<EmailClient>
+) -> Result<HttpResponse, CustomError> {
+    let send_appointment_emails = send_appointment_emails.into_inner();
+    let mut payload: Vec<NewInvitation>  = vec![];
+    let appt_id = appointment_id.into_inner();
+    let qr_client = qr_client.into_inner();
+    let email_client = email_client.into_inner();
+
+    let count = match &send_appointment_emails.email {
+        None => { query_params.count.unwrap_or(1) }
+        Some(emails) => {
+            emails.len() as i32
+        }
+    };
+
+    let futures: Vec<_> = (0..count).map(|_| {
+        let id = Uuid::new_v4();
+        let qr_client_clone = qr_client.clone();
+        async move {
+            match qr_client_clone.get_short_url(id.to_string()).await {
```

Comment:
Try this:
qr_client_clone.get_short_url(id.to_string()).await.map(|res| (id, res.short_url))